sql/src/main/java/io/crate/executor/transport/TableCreator.java

```diff
@@ -40,71 +41,78 @@
 import org.elasticsearch.action.admin.indices.template.put.PutIndexTemplateResponse;
 import org.elasticsearch.action.admin.indices.template.put.TransportPutIndexTemplateAction;
 import org.elasticsearch.cluster.ClusterService;
+import org.elasticsearch.common.logging.ESLogger;
+import org.elasticsearch.common.logging.Loggers;
+import org.elasticsearch.common.settings.Settings;
 import org.elasticsearch.indices.IndexAlreadyExistsException;
 import org.elasticsearch.indices.IndexTemplateAlreadyExistsException;
 
-import java.util.List;
 import java.util.Locale;
-import java.util.UUID;
 
-public class CreateTableTask extends AbstractChainedTask {
+public class TableCreator {
 
-    private static final TaskResult SUCCESS_RESULT = TaskResult.ONE_ROW;
+    protected static final ESLogger logger = Loggers.getLogger(TableCreator.class);
 
     private final ClusterService clusterService;
     private final TransportCreateIndexAction createIndexAction;
     private final TransportDeleteIndexAction deleteIndexAction;
     private final TransportPutIndexTemplateAction putIndexTemplateAction;
-    private final CreateTableNode planNode;
-
-    public CreateTableTask(UUID jobId,
-                           ClusterService clusterService,
-                           TransportCreateIndexAction createIndexAction,
-                           TransportDeleteIndexAction deleteIndexAction, TransportPutIndexTemplateAction putIndexTemplateAction,
-                           CreateTableNode node) {
-        super(jobId);
+    private final CreateTableAnalyzedStatement statement;
+
+    private final Settings settings;
+
+    public TableCreator(ClusterService clusterService,
+                        TransportCreateIndexAction createIndexAction,
+                        TransportDeleteIndexAction deleteIndexAction,
+                        TransportPutIndexTemplateAction putIndexTemplateAction,
+                        CreateTableAnalyzedStatement statement) {
         this.clusterService = clusterService;
         this.createIndexAction = createIndexAction;
         this.deleteIndexAction = deleteIndexAction;
         this.putIndexTemplateAction = putIndexTemplateAction;
-        this.planNode = node;
+        this.statement = statement;
+
+        this.settings = statement.tableParameter().settings().getByPrefix("index.");
     }
 
-    @Override
-    protected void doStart(List<TaskResult> upstreamResults) {
+
+    public ListenableFuture<Void> start() {
```

Comment:
can the return value now be changed so that the wrapRowCountFuture stuff isn't necessary?